src/main/java/org/broadinstitute/hellbender/tools/exome/cnlohcaller/CNLOHCaller.java

```diff
@@ -0,0 +1,803 @@
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.serializer.KryoSerializer;
+import org.apache.spark.serializer.SerializerInstance;
+import org.broadinstitute.hellbender.tools.exome.ACNVModeledSegment;
+import org.broadinstitute.hellbender.utils.GATKProtectedMathUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+import scala.reflect.ClassTag;
+import scala.reflect.ClassTag$;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.SortedMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+
+/**
+ * <p>This class makes calls for each segment on whether the segment is CNLoH and/or balanced (MAF=0.5).  Please see the
+ *  docs for detailed information.
+ * </p>
+ *
+ * <p>Throughout this class, rho is the CCF * purity.  M is the copy number of one allele and N is the copy number of the other allele.
+ * There is no assumption about the relative relationship of M and N, though typically, we try to maintain M >= N,
+ *  though this is not guaranteed.
+ *</p>
+ *
+ *  <p>Many of the variable names assume that you have read the CNV methods pdf in the docs directory.</p>
+ *
+ * Variables:
+ * <ul>
+ *     <li>rho -- CCF * purity.  We use a set number of rhos that each segment can have.</li>
+ *     <li>M -- absolute copy number for allele 1</li>
+ *     <li>N -- absolute copy number for allele 2.  NOT assumed to have same or fewer copies than allele 1</li>
+ *     <li>pi -- discrete pdf for values of M and N</li>
+ *     <li>phi -- likelihood of each rho value</li>
+ *     <li>maf -- minor allelic fraction</li>
+ *     <li>fMaf -- likelihood of a minor allelic fraction, given other parameters</li>
+ *     <li>cr -- copy ratio</li>
+ *     <li>fCr -- likelihood of a copy ratio, given other parameters</li>
+ *     <li>alpha -- concentration parameter of dirichlet prior on z_s</li>
+ *     <li>z_s -- indicator variable (binary) whether segment s has a CNV on segment s.  This is not explicitly used in the code.</li>
+ * </ul>
+ *
+ *  NOTES:
+ *  <ul>
+ *      <li>{@code lambda} (CCF * ploidy) is present, but the value is never changed.  In future versions, we may add this to the model.</li>
+ *  </ul>
+ */
+public class CNLOHCaller implements Serializable {
+
+    private static final Logger logger = LogManager.getLogger(CNLOHCaller.class);
+    protected static double MIN_L = 1e-50;
+    protected static double MIN_DIST_FROM_MODE = 1e-10;
+    static final long serialVersionUID = 337337337123L;
+
+    private double segmentMeanBiasInCR;
+    private double segmentMeanVarianceInCR;
+
+    private double rhoThreshold;
+
+    /** How many copies expected in a normal.  For humans, this is 2 */
+    private int normalNumCopies;
+
+    public CNLOHCaller(){
+        rhoThreshold = 0.2;
```

Comment:
magic constant.  Extract a static constant and make it public in order to reference it in the CLI.